pkg/remotewrite/remotewrite.go

```diff
@@ -80,11 +89,51 @@ func (o *Output) Start() error {
 
 func (o *Output) Stop() error {
 	o.logger.Debug("Stopping the output")
+	defer o.logger.Debug("Output stopped")
 	o.periodicFlusher.Stop()
-	o.logger.Debug("Output stopped")
+
+	staleMarkers := o.staleMarkers(time.Now())
+	o.logger.WithField("staleMarkers", len(staleMarkers)).Debug("Marking time series as stale, if any")
+	if len(staleMarkers) < 1 {
+		return nil
+	}
+
+	// TODO: warn if it the request is taking too long (5 seconds?, the same timeout set for flushing)
+
+	err := o.client.Store(context.Background(), staleMarkers)
+	if err != nil {
+		return fmt.Errorf("marking time series as stale failed: %w", err)
+	}
 	return nil
 }
 
+// staleMarkers maps all the seen time series with a stale marker.
+func (o *Output) staleMarkers(t time.Time) []*prompb.TimeSeries {
+	timestamp := t.UnixMilli()
+	staleMarkers := make([]*prompb.TimeSeries, 0, len(o.tsdb))
+
+	for _, swm := range o.tsdb {
```

Comment:
Could this race with any writes to `o.tsdb`?

Reply:
`tsdb` is used by `flushMetrics` and `Stop`, `Stop` calls `o.periodicFlusher.Stop()` that should guarantee no more invocations for `flushMetrics`, so I don't think it can be racy.

The `periodicFlusher` core code:

https://github.com/grafana/k6/blob/4e4e5b1ff730ce4fcce8f8296b8fbb767ec429d9/output/helpers.go#L63-L86

Reply:
Thanks for confirming :+1: